scripts/overton/EDA_script.py

```diff
@@ -0,0 +1,176 @@
+    df['n_words'].hist()
+
+    count_language = Counter(spacy_df['paragraph_language'][0])
+    for i in range(1, len(spacy_df)):
+        count_language += Counter(spacy_df['paragraph_language'][i])
+    count_language = eda.count_texts(count_language)
+    count_language.to_excel('excel/count_language.xlsx')
+    count_language
+
+    nouns = []
+    for i in range(len(spacy_df['merged_nouns'])):
+        nouns = np.concatenate((nouns,spacy_df['merged_nouns'][i]))
+    lemmas = []
+    for i in range(len(spacy_df['merged_lemmas'])):
+        lemmas = np.concatenate((lemmas,spacy_df['merged_lemmas'][i]))
+    noun_chunks = []
+    for i in range(len(spacy_df['merged_noun_chunks'])):
+        noun_chunks = np.concatenate((noun_chunks,spacy_df['merged_noun_chunks'][i]))
+
+    with open("EDA_report.txt", "a") as file_object:
+        file_object.write('Number of nouns:'+ str(len(nouns)) + '\n')
+        file_object.write('Number of lemmas:'+ str(len(lemmas)) + '\n')
+        file_object.write('Number of noun_chunks:'+ str(len(noun_chunks)) + '\n')
+
+    print('Number of nouns:', len(nouns))
+    print('Number of lemmas:', len(lemmas))
+    print('Number of noun_chunks:', len(noun_chunks))
+
+    count_nouns = eda.count_texts(nouns)
+    eda.plot_counts(count_nouns, ['obj', 'count']).write_image(file = 'plots/bar_counted_nouns.png', format = 'png')
+    eda.plot_counts(count_nouns, ['obj', 'count'])
+
+    word_counts = Counter(lemmas)
+    wc = WordCloud(width=1600, height=800)
+    wc.generate_from_frequencies(frequencies = word_counts)
+    plt.figure(figsize = (18, 14))
+    plt.imshow(wc)
+    plt.savefig('plots/word_cloud.png')
+
+    count_lemmas= eda.count_texts(lemmas)
+    eda.plot_counts(count_lemmas, ['obj', 'count']).write_image(file = 'plots/bar_counted_lemmas.png', format = 'png')
+    eda.plot_counts(count_lemmas, ['obj', 'count'])
+
+    stopwords = en.Defaults.stop_words
+    top_n_bigrams=eda.get_top_ngram(df['Text_all'], stopwords = stopwords, n = 2, m = 2)[:30]
+    x, y=map(list,zip(*top_n_bigrams))
+    fig = plt.subplots(figsize=(15, 15))
+    sns.barplot(x = y, y = x)
+    plt.savefig('plots/bigram.png')
+
+    top_n_bigrams=eda.get_top_ngram(df['Text_all'], stopwords = stopwords , n = 3, m = 3)[:30]
+    x, y=map(list,zip(*top_n_bigrams))
+    fig = plt.subplots(figsize=(15, 15))
+    sns.barplot(x = y, y = x)
+    plt.savefig('plots/trigram.png')
+
+    top_n_bigrams=eda.get_top_ngram(df['Text_all'], stopwords = stopwords, n = 4, m = 4)[:30]
+    x, y=map(list,zip(*top_n_bigrams))
+    fig = plt.subplots(figsize=(15, 15))
+    sns.barplot(x = y, y = x)
+    plt.savefig('plots/fourgram.png')
+
+    noun_chunks = list(filter(lambda x: len(x.split()) > 1, noun_chunks))
+    count_chunks = eda.count_texts(noun_chunks,['chunk', 'count'], 40)
+    eda.plot_counts(count_chunks, ['chunk', 'count']).write_image(file = 'plots/bar_counted_chunks_pt1.png', format = 'png')

```

Comment:
tu może jeszcze mały fix, pt1, pt2, pt3 jest trochę mało interpretowalne jak się nie widziało kodu, zróbmy może file = f'plots/bar_counted_chunks_{tu jakieś paramsy dzięki którym łatwiej będzie ogarnąć który to plik}.png'